pkg/webhook/clustersetbinding/validating_webhook.go

```diff
@@ -0,0 +1,145 @@
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+)
+
+// ManagedClusterSetBindingValidatingAdmissionHook will validate the creating/updating ManagedClusterSetBinding request.
+type ManagedClusterSetBindingValidatingAdmissionHook struct {
+	kubeClient    kubernetes.Interface
+	clusterClient clusterclient.Interface
+}
+
+// ValidatingResource is called by generic-admission-server on startup to register the returned REST resource through which the
+// webhook is accessed by the kube apiserver.
+func (a *ManagedClusterSetBindingValidatingAdmissionHook) ValidatingResource() (plural schema.GroupVersionResource, singular string) {
+	return schema.GroupVersionResource{
+			Group:    "admission.cluster.open-cluster-management.io",
+			Version:  "v1",
+			Resource: "managedclustersetbindingvalidators",
+		},
+		"managedclustersetbindingvalidators"
+}
+
+// Validate is called by generic-admission-server when the registered REST resource above is called with an admission request.
+func (a *ManagedClusterSetBindingValidatingAdmissionHook) Validate(admissionSpec *admissionv1beta1.AdmissionRequest) *admissionv1beta1.AdmissionResponse {
+	klog.V(4).Infof("validate %q operation for object %q", admissionSpec.Operation, admissionSpec.Object)
+
+	status := &admissionv1beta1.AdmissionResponse{}
+
+	// only validate the request for ManagedClusterSetBinding
+	if admissionSpec.Resource.Group != "cluster.open-cluster-management.io" ||
+		admissionSpec.Resource.Resource != "managedclustersetbindings" {
+		status.Allowed = true
+		return status
+	}
+
+	switch admissionSpec.Operation {
+	case admissionv1beta1.Create, admissionv1beta1.Update:
+		return a.validateRequest(admissionSpec)
+	default:
+		status.Allowed = true
+		return status
+	}
+}
+
+// Initialize is called by generic-admission-server on startup to setup initialization that ManagedClusterSetBinding webhook needs.
+func (a *ManagedClusterSetBindingValidatingAdmissionHook) Initialize(kubeClientConfig *rest.Config, stopCh <-chan struct{}) error {
+	var err error
+	a.kubeClient, err = kubernetes.NewForConfig(kubeClientConfig)
+	if err != nil {
+		return err
+	}
+
+	a.clusterClient, err = clusterclient.NewForConfig(kubeClientConfig)
+	return err
+}
+
+// validateCreateRequest validates the create/update operation of ManagedClusterSetBinding
+func (a *ManagedClusterSetBindingValidatingAdmissionHook) validateRequest(request *admissionv1beta1.AdmissionRequest) *admissionv1beta1.AdmissionResponse {
+	status := &admissionv1beta1.AdmissionResponse{}
+
+	binding := &clusterv1alpha1.ManagedClusterSetBinding{}
+	if err := json.Unmarshal(request.Object.Raw, binding); err != nil {
+		status.Allowed = false
+		status.Result = &metav1.Status{
+			Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+			Message: err.Error(),
+		}
+		return status
+	}
+
+	// check if target cluster set is not specified
+	if len(binding.Spec.ClusterSet) == 0 {
```

Comment:
you can do this much in CRD schemas.  Get up with mdame in slack or #forum-api-review

Reply:
Removed the checking and added a validation rule on CRD.